test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java

```diff
@@ -108,8 +109,7 @@ public void doTestActions(String targetVMId) throws Throwable {
                     "VirtualMachineDescriptor.equals() returns 'false' for '" + targetVMDesc + "' and '" + targetVMDesc2 + "'");
 
             TestUtils.assertEquals(targetVMDesc.hashCode(), targetVMDesc2.hashCode(),
-                    "VirtualMachineDescriptor.hashCode() returns different values " + "(" + targetVMDesc.hashCode() + " and " + targetVMDesc2.hashCode() + ")" +
-                    " for '" + targetVMDesc + "' and '" + targetVMDesc2 + "'");
+                    "VirtualMachineDescriptor.hashCode() returns different values for '" + targetVMDesc + "' and '" + targetVMDesc2 + "'");
```

Comment:
Does `TestUtils.assertEquals` print out actual/expected value as well? If not, dropping `hashCode` printout here loses debugging data.

Reply:
it does for both a reference type and `int`, [TestUtils.java]:
```
    public static void assertEquals(int i1, int i2) {
        if (i1 != i2) {
            throw new TestFailure(
                    format("Check failed: %d != %d", i1, i2));
        }
    }

    public static void assertEquals(Object obj1, Object obj2, Object errorMessage) {
        assertTrue(obj1.equals(obj2), new LazyFormatString("%s: [%s] != [%s]", errorMessage,  obj1, obj2));
    }
```
[TestUtils.java]: https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/vmTestbase/nsk/share/test/TestUtils.java#L70

Reply:
OK, cool.